agent/bench-scripts/driver/linpack

```diff
@@ -0,0 +1,277 @@
+			run_trials=${1//,/ }
+			shift
+		fi
+		;;
+	--alignment-values)
+		# Values for line 7 of the LINPACK input file
+		if [[ -n "${1}" ]]; then
+			alignment_values=${1//,/ }
+			shift
+		fi
+		;;
+	--threads)
+		if [[ -n "${1}" ]]; then
+			threads_arg="${1}"
+			shift
+		fi
+		;;
+	--use-omp)
+		if [[ -n "${1}" ]]; then
+			use_omp="${1}"
+			shift
+		fi
+		;;
+	--kmp-affinity)
+		if [[ -n "${1}" ]]; then
+			kmp_affinity_args="${1}"
+			shift
+		fi
+		;;
+	--numactl-args)
+		if [[ -n "${1}" ]]; then
+			numactl_args="${1}"
+			shift
+		fi
+		;;
+	--output-dir)
+		if [[ -n "${1}" ]]; then
+			output_dir="${1}"
+			shift
+		fi
+		;;
+	-h|--help)
+		usage
+		exit 0
+		;;
+	--)
+		break
+		;;
+	*)
+		printf -- "[%s] WARNING: unrecognized parameter encountred, '%s'\n" "${script_name}" "${arg}" >&2
+		break
+		;;
+	esac
+done
+
+if [[ -z "${output_dir}" || ! -d "${output_dir}" ]]; then
+	printf -- "[%s] ERROR: Specified --output-dir '${output_dir}' is not a directory\n" "${script_name}" >&2
+	exit 2
+fi
+if [[ -z "${binary}" ]]; then
+	printf -- "[%s] ERROR: You must specify --binary\n" "${script_name}" >&2
+	exit 3
+fi
+if [[ ! -e "${binary}" || ! -x "${binary}" ]]; then
+	printf -- "[%s] ERROR: The --binary must exist and be executable\n" "${script_name}" >&2
+	exit 4
+fi
+if [[ "${use_omp}" != "y" && "${use_omp}" != "n" ]]; then
+	printf -- "[%s] ERROR: --use-omp must either by 'y' or 'n'\n" "${script_name}" >&2
+	exit 5
+fi
+num_problem_sizes=$(echo "${problem_sizes}" | wc -w)
+num_leading_dimensions=$(echo "${leading_dimensions}" | wc -w)
+num_run_trials=$(echo "${run_trials}" | wc -w)
+num_alignment_values=$(echo "${alignment_values}" | wc -w)
+if [[ "${num_problem_sizes}" != "${num_leading_dimensions}" || "${num_leading_dimensions}" != "${num_run_trials}" || "${num_run_trials}" != "${num_alignment_values}" ]]; then
+	printf -- "[%s] ERROR: You must have the same number of elements for --problem-sizes, --leading-dimensions, --run-trials, and --alignment-values\n" >&2
+	exit 6
+fi
+number_of_tests=${num_problem_sizes}
+
+linpack_input_file="${output_dir}/${linpack_input_file_name}"
+linpack_metadata_file="${output_dir}/${linpack_metadata_file_name}"
+linpack_output_file="${output_dir}/${linpack_output_file_name}"
+
+numactl_cmd=""
+if [[ -n "${numactl_args}" ]]; then
+	# N.B. - Trailing whitespace is required.
+	numactl_cmd="numactl ${numactl_args} "
+fi
+
+if [[ "${use_omp}" == "y" ]]; then
+	if [[ -z "${threads_arg}" ]]; then
+		threads_arg=$(cat /proc/cpuinfo | grep processor | wc -l)
+	fi
+	# N.B. - Trailing whitespace is required.
+	omp_num_threads="OMP_NUM_THREADS=${threads_arg} "
+	# N.B. - Trailing whitespace is required.
+	kmp_affinity="KMP_AFFINITY=${kmp_affinity_args} "
+else
+	# When "use OpenMP" is false, we are not going to use the "threads_arg" or "kmp_affinity_args"
+	threads_arg=""
+	kmp_affinity_args=""
+
+	# No environment variables to add to the command line.
+	omp_num_threads=""
+	kmp_affinity=""
```

Comment:
I kept them here next to their use in `final_cmd` since that is their only purpose, they are not part of the CLI interface.

I don't think the reduction is code is helpful here because we want to call the attention of the reader to the semantic that when `use_omp` is not `y`, we don't use either of those values.

Reply:
OK, given my [recent revelation](https://github.com/distributed-system-analysis/pbench/pull/2768#discussion_r856733848), I need to amend my statement above:  in the current code structure, we _would_ need to clear `kmp_affinity_args` here.  😝  So, let me start over....

There are two obvious (to me...) approaches to structuring code like this.

The first declares a set of variables and initializes them using a coherent set of defaults; it then collects the command line arguments, overriding the variables' values as specified; finally, it uses the variables for the job at hand.

The second declares a set of more-or-less constant values indicating what the default values are; it then collects the command line arguments in a different set of variables; it then uses the variables for the job at hand, filling in the defaults for any variables which aren't set by the command line.

Unfortunately, the code here is doing something sort of in between.

That is, it's mostly following the first approach above, and we're sort of set up for `use_omp == "y"` by default.  If that setup were complete, then we wouldn't need to change any options unless we find that `use_omp == "n"`.  However, we didn't initialize `threads_arg` properly, and we defer setting `omp_num_threads` and `kmp_affinity` until we're done with the command line processing.  (The latter two are more or less necessary, but I think it would be better if we put the `/proc/cpuinfo` count up at line 72 so that we don't need to touch it at line 223 and we can just use it as is to set up `omp_num_threads`.)

The weird thing is, in the `use_omp == "n"` case, we're still putting the `threads` and `kmp_affinity` keys in the metadata, even though their values are _blank_.  It seems like we should be omitting them, in which case we wouldn't have to worry about clearing their values in the code referenced here.

Reply:
> That is, it's mostly following the first approach above, and we're sort of set up for `use_omp == "y"` by default. If that setup were complete, then we wouldn't need to change any options unless we find that `use_omp == "n"`. However, we didn't initialize `threads_arg` properly, and we defer setting `omp_num_threads` and `kmp_affinity` until we're done with the command line processing. (The latter two are more or less necessary, but I think it would be better if we put the `/proc/cpuinfo` count up at line 72 so that we don't need to touch it at line 223 and we can just use it as is to set up `omp_num_threads`.)

The previous behavior of `linpack-prepare-input-files` only fetched the CPU count from `/proc/cpuinfo` when `${use_omp} == "y"` and no explicit `threads` value was provided.  So this code maintains that behavior.

I added use of `_def` values to make it clearer when we end up using defaults.

> 
> The weird thing is, in the `use_omp == "n"` case, we're still putting the `threads` and `kmp_affinity` keys in the metadata, even though their values are _blank_. It seems like we should be omitting them, in which case we wouldn't have to worry about clearing their values in the code referenced here.

The previous behavior of `linpack-prepare-input-files` did that, so I kept [that behavior](https://github.com/distributed-system-analysis/pbench/blob/main/agent/bench-scripts/postprocess/linpack-prepare-input-file#L220).